source/geometry_model/chunk.cc

```diff
@@ -0,0 +1,416 @@
+
+          prm.declare_entry ("Chunk minimum longitude", "0",
+                             Patterns::Double (-180, 360), // enables crossing of either hemisphere
+                             "Minimum longitude of the chunk. Units: degrees.");
+          prm.declare_entry ("Chunk maximum longitude", "1",
+                             Patterns::Double (-180, 360), // enables crossing of either hemisphere
+                             "Maximum longitude of the chunk. Units: degrees.");
+
+          prm.declare_entry ("Chunk minimum latitude", "0",
+                             Patterns::Double (-90, 90),
+                             "Minimum latitude of the chunk. This value is ignored "
+                             "if the simulation is in 2d. Units: degrees.");
+          prm.declare_entry ("Chunk maximum latitude", "1",
+                             Patterns::Double (-90, 90),
+                             "Maximum latitude of the chunk. This value is ignored "
+                             "if the simulation is in 2d. Units: degrees.");
+
+          prm.declare_entry ("Radius repetitions", "1",
+                             Patterns::Integer (1),
+                             "Number of cells in radius.");
+          prm.declare_entry ("Longitude repetitions", "1",
+                             Patterns::Integer (1),
+                             "Number of cells in longitude.");
+          prm.declare_entry ("Latitude repetitions", "1",
+                             Patterns::Integer (1),
+                             "Number of cells in latitude. This value is ignored "
+                             "if the simulation is in 2d");
+
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+
+
+    template <int dim>
+    void
+    Chunk<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Geometry model");
+      {
+        prm.enter_subsection("Chunk");
+        {
+
+          const double degtorad = dealii::numbers::PI/180;
+
+          Assert (dim >= 2, ExcInternalError());
+          Assert (dim <= 3, ExcInternalError());
+
+          if (dim >= 2)
+            {
+              point1[0] = prm.get_double ("Chunk inner radius");
+              point2[0] = prm.get_double ("Chunk outer radius");
+              repetitions[0] = prm.get_integer ("Radius repetitions");
+              point1[1] = prm.get_double ("Chunk minimum longitude") * degtorad;
+              point2[1] = prm.get_double ("Chunk maximum longitude") * degtorad;
+              repetitions[1] = prm.get_integer ("Longitude repetitions");
+
+              Assert (point1[0] < point2[0],
+                      ExcMessage ("Inner radius must be less than outer radius."));
+              Assert (point1[1] < point2[1],
+                      ExcMessage ("Minimum longitude must be less than maximum longitude."));
+              Assert (point2[1] - point1[1] < 2.*numbers::PI,
```

Comment:
Please use AssertThrow in all places where you check input parameters. Assert gets optimized out in release mode, and people will continue to use wrong parameter files even in release mode.